### DISCUSSION TITLE
Is it possible to avoid compiling the package when installing?

### DISCUSSION THREAD

**Discussion by markis** [ofek/hatch-mypyc#35] [Q&A] [answered]
As the title says, is it possible to avoid compiling the package when installing?

For instance, when installing the dev optional dependencies, I would use this `pip install -e '.[dev]'` and it would install the dev dependencies.  But it would still build and compile the package.

**Comment by ofek** [comment, marked as answer]
https://hatch.pypa.io/latest/config/build/#conditional-execution

**Reply by markis** [reply]
Ah, I see!  The way I tried and seem to work was to disable mypyc by default.

```toml
[tool.hatch.build.targets.wheel.hooks.mypyc]
enable-by-default = false
```

Then when I need the project compiled, I can use:

```shell
HATCH_BUILD_HOOK_ENABLE_MYPYC=1 pip install `.[dev]`
```

**Reply by markis** [reply]
Thanks @ofek !

**Reply by ofek** [reply]
Happy to help!
